### PR TITLE
Sitemap enhancements & URL fixes

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <mehranb@amazon.com>

### Description
* Improves sitemap's last-modified
* Fixes paths to `versions`, `artifacts`, and `authors`
* Lets robots know to not bother with `javadocs`
 
### Issues Resolved
[List any issues this PR will resolve]

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -54,13 +55,13 @@ defaults:
       type: versions
     values: 
       layout: versions
-      permalink:  versions/:categories/:slug
+      permalink:  versions/:categories/:slug.html
```
Comment (stockholmux) [inline _config.yml:58]: Good catch on these @AMoo-Miki.  Is there an SEO advantage to using raw paths (`/versions/foo/bar/`) vs files (`/versions/foo/bar.html`)? The site is pretty inconsistent when I step back and look at it.

File: Gemfile [hunk 2]
```diff
@@ -17,6 +17,7 @@ gem "minima", "~> 2.5"
 group :jekyll_plugins do
   gem "jekyll-feed", "~> 0.12"
   gem "jekyll-paginate"
+  gem "jekyll-last-modified-at"
```
Comment (stockholmux) [inline Gemfile:20]: what does this plugin do? Why is it needed?

**Review by stockholmux — COMMENTED** [review]
A few questions...

File: _config.yml (hunk 1 above)
Comment (AMoo-Miki) [reply _config.yml:58]: SEO-wise, none. The only reason I chose the `.html` was because the permalinks that visitors might have shared.

Comment (stockholmux) [reply _config.yml:58]: Makes sense - given the name of the PR branch, I just wanted to confirm that we weren't doing anything counter to SEO best practices.

File: Gemfile (hunk 2 above)
Comment (stockholmux) [reply Gemfile:20]: OK. Makes sense - if it helps the site map 👍

**Review by stockholmux — APPROVED** [review]
LGTM

**Pull request merged**